quafu/visualisation/draw_dag.py

```diff
@@ -0,0 +1,55 @@
+import graphviz
+
+from quafu import QuantumCircuit
+from quafu.dagcircuits.circuit_dag import circuit_to_dag
+from quafu.dagcircuits.instruction_node import InstructionNode
+
+from typing import Union, Any
+
+
+def _extract_node_info(node):
+    if isinstance(node, InstructionNode):
+        name, label = str(id(node)), str(node)
+    else:
+        assert node == -1 or node == float("inf")
+        name, label = str(node), str(node)
+    return name, label
+
+
+def draw_dag(qc: Union[QuantumCircuit, None],
+             dag: Any = None,
+             output_format: str = 'pdf',
+             output_filename: str = 'DAG'):
+    """
+    TODO: complete docstring, test supports for notebook
+
+        Helper function to visualize the DAG
+
+        Args:
+            qc (QuantumCircuit): pyquafu quantum circuit if provided
+            dag (DAG): DAG object with nodes and edges, built from qc if not provided
+            output_format (str): output format, including "png", "svg", "pdf"...
+            output_filename (str): file name of generated image
+
+        Returns:
+            dot: graphviz.Digraph object
+
+        """
```

Comment:
Just a minor issue, align the indent ^_^

Reply:
Yes, the docstring needs updating as well. I will finish it later.